File: problem set 1/mario.c
#include <stdio.h>
#include <cs50.h>


int main(void)
{
    // Get the number of blocks from the user
    int height = -1;
    while (height < 0 || height > 23)
    {
        printf("Height (0-23): ");
        height = get_int();
    }

    // Large loop for each row, nested loops for characters within the row
    int i, j;
    for (i = height; i >= 1; i--)
    {
        for (j = i - 1; j > 0; j--)         // spaces before the hashes before the repeating block
            printf(" ");

        for (j = 0; j < height - i; j++)    // hashes before the repeating block
            printf("#");

        printf("#  #");                     // central, repeating block

        for (j = 0; j < height - i; j++)    // hashes after the repeating block
            printf("#");

        printf("\n");                       // newline before starting next tow
    }

    return 0;
}

File: problem set 6/sentimental/mario.py

# Get the number of blocks from the user, must be between 0 and 23, both included
while True:
    try:
        height = -1     # force the first while loop
        while (int(height) < 0 or int(height) > 23):
            height = input("Height (0-23): ")
            height = int(height)

            if height < 0 or height > 23:
                print("Enter a number between 0 and 23, both included")

        break
    except ValueError:
        print("Enter a number")


# Large loop for each row, nested loops for characters within the row
for i in range(height, 0, -1):

    for j in range(i-1, 0, -1):         # spaces before the hashes before the repeating block
        print(" ", end="")

    for j in range(0, height-i):        # hashes before the repeating block
        print("#", end="")

    print("#  #", end="")               # central, repeating block

    for j in range(0, height-i):        # hashes after the repeating block
        print("#", end="")

    print()                             # newline before starting next tow
